tgui/packages/tgui/interfaces/PrisonerShuttleConsole.js

```diff
@@ -0,0 +1,64 @@
+import { useBackend } from '../backend';
+import {
+  Button,
+  ProgressBar,
+  LabeledList,
+  Section,
+} from '../components';
+import { Window } from '../layouts';
+
+export const PrisonerShuttleConsole = (props, context) => {
+  const { act, data } = useBackend(context);
+  const {
+    can_go_home,
+    emagged,
+    id_inserted,
+    id_name,
+    id_points,
+    id_goal,
+  } = data;
+  const bad_progress = emagged ? 0 : 1;
+  const completionStatus = emagged
+    ? 'ERR0R'
+    : can_go_home
+    ? 'Completed!'
+    : 'Insufficient';
+  return (
+    <Window>
+      <Window.Content>
+        <LabeledList>
+          <LabeledList.Item label="Status">
+            {(!!id_inserted && (
+              <ProgressBar
+                value={id_points / id_goal}
+                ranges={{
+                  good: [bad_progress, Infinity],
+                  bad: [-Infinity, bad_progress],
+                }}
+              >
+                {id_points + ' / ' + id_goal + ' ' + completionStatus}
+              </ProgressBar>
+            )) ||
+              (!!emagged && 'ERR0R COMPLETED?!@') ||
+              'No ID inserted'}
```

Comment:
This could benefit from having all this logic done outside the return block.

```js
const statusBlock = 'No ID inserted'
if (id_inserted) {
  statusBlock = (
    <ProgressBar
        value={id_points / id_goal}
        ranges={{
          good: [bad_progress, Infinity],
          bad: [-Infinity, bad_progress],
        }}>
        {id_points + ' / ' + id_goal + ' ' + completionStatus}
      </ProgressBar>)
}
else if (emagged) {
    statusBlock = 'ERR0R COMPLETED?!@')
}
```
then all you would need to do is
```js
<LabeledList.Item label="Status">
    {statusBlock}
</LabeledList.Item>

Reply:
same here

Reply:
see comment on above review about using `let` instead of `const`